internal/k8shandler/status.go

```diff
@@ -284,76 +284,66 @@ func (er *ElasticsearchRequest) updateNodeConditions(status *api.ElasticsearchSt
 
 func (er *ElasticsearchRequest) updateStorageConditions(status *api.ElasticsearchStatus) error {
 	ll := er.L()
-	client := er.client
-	cluster := er.cluster
 
-	for _, node := range cluster.Spec.Nodes {
+	emptySpecVol := api.ElasticsearchStorageSpec{}
+	structureStatus, nameStatus, sizeStatus := v1.ConditionFalse, v1.ConditionFalse, v1.ConditionFalse
+
+	nodeNames := []string{}
+	clusterNodes := nodes[nodeMapKey(er.cluster.GetName(), er.cluster.GetNamespace())]
+
+	for _, node := range clusterNodes {
+		nodeNames = append(nodeNames, node.name())
+	}
+
+	for _, node := range er.cluster.Spec.Nodes {
+		nodeName := nodeNameContains(node.GenUUID, nodeNames)
+
+		if nodeName == "" {
+			ll.Info("Unable to find appropriate node to compare storage spec.")
+			continue
+		}
+
 		specVol := node.Storage
 		current := &v1.PersistentVolumeClaim{}
-		emptySpecVol := api.ElasticsearchStorageSpec{}
+		claimName := fmt.Sprintf("%s-%s", er.cluster.Name, nodeName)
+
+		isUsingPVCStorageSpec := true
+		isEphemeralStorageSpec := reflect.DeepEqual(specVol, emptySpecVol) || specVol.Size == nil
 
-		// When these conditions are not met, ephemeral storage
-		// is used instead of making a pvc claim.
-		didMakePVCClaim := true
-		willMakePVCClaim := !(reflect.DeepEqual(specVol, emptySpecVol) || specVol.Size == nil)
+		if isEphemeralStorageSpec {
+			ll.Info("iS eph")
+		}
```

Comment:
is this just a debug line that'll be removed before the final commit?

Reply:
Yes. I could've sworn I had removed those before push this up.